### PR TITLE
testing/py-pillow-simd: new aport

### PR DESCRIPTION
**Pull request by messer**
Provides the SSE4 vectorized version of Pillow-SIMD from UploadCare, as an
alternative to py-pillow.

### CODE REVIEW

**Comment by maxice8** [comment]
Please don't add py2- stuff, have the package be py3-pillow-simd

**Comment by messer** [comment]
Thank you for reminding me of the futility of supporting 2.x at this point! Changes made as requested.

**Comment by maxice8** [comment]
On Tue, 10 Sep 2019 17:27:09 -0700
Marcel Esser <notifications@github.com> wrote:

> Thank you for reminding me of the futility of supporting 2.x at this
> point! Changes made as requested.
> 


No worries

File: testing/py3-pillow-simd/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+pkgrel=0
+pkgdesc="Python Imaging Library SIMD"
+url="https://github.com/uploadcare/pillow-simd"
+arch="x86_64"
+license="PSF"
+provides="py-pillow py3-pillow"
+depends="py3-olefile freetype jpeg libwebp tiff libpng lcms2 libjpeg-turbo zlib"
+makedepends="python3-dev py3-setuptools freetype-dev jpeg-dev libwebp-dev
+	tiff-dev libpng-dev lcms2-dev libjpeg-turbo-dev zlib-dev"
+subpackages="py3-${pkgname#py-}:_py3"
```
Comment (maxice8) [inline testing/py3-pillow-simd/APKBUILD]: use $_pkgname

File: testing/py3-pillow-simd/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Maintainer: Marcel Esser <marcel.esser@gmail.com>
+pkgname=py3-pillow-simd
+_pkgname=Pillow-SIMD
+pkgver=6.0.0
+pkgrel=0
+pkgdesc="Python Imaging Library SIMD"
+url="https://github.com/uploadcare/pillow-simd"
+arch="x86_64"
+license="PSF"
+provides="py-pillow py3-pillow"
```
Comment (maxice8) [inline testing/py3-pillow-simd/APKBUILD]: I'd rather have people explicitly install this package

File: testing/py3-pillow-simd/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+depends="py3-olefile freetype jpeg libwebp tiff libpng lcms2 libjpeg-turbo zlib"
+makedepends="python3-dev py3-setuptools freetype-dev jpeg-dev libwebp-dev
+	tiff-dev libpng-dev lcms2-dev libjpeg-turbo-dev zlib-dev"
+subpackages="py3-${pkgname#py-}:_py3"
+source="https://files.pythonhosted.org/packages/b1/19/b7043190f481abb94dcdd1e69c4432432aaa73455cf1128eae39b8eb2518/Pillow-SIMD-$pkgver.post0.tar.gz"
```
Comment (maxice8) [inline testing/py3-pillow-simd/APKBUILD]: please use a simpler url like other py3- packages do

File: testing/py3-pillow-simd/APKBUILD (hunk 1 above)
Comment (messer) [reply testing/py3-pillow-simd/APKBUILD]: Resolved in 2adbbd8fe9a12735d28542865a16663851bde0fd

File: testing/py3-pillow-simd/APKBUILD (hunk 2 above)
Comment (messer) [reply testing/py3-pillow-simd/APKBUILD]: Resolved in 2adbbd8fe9a12735d28542865a16663851bde0fd

File: testing/py3-pillow-simd/APKBUILD (hunk 3 above)
Comment (messer) [reply testing/py3-pillow-simd/APKBUILD]: I believe I have resolved this with 2adbbd8fe9a12735d28542865a16663851bde0fd. Please note that this PyPi package uses a "post-release" version number which is defined by PyPi conventions, but I have never actually seen before in any package. There also appear to be no examples of its use in existing Alpine packages. Its is analogous to Alpine's pkgrel incrementor. I decided to resolve this by adding an underscored variable to the APKBUILD file to represent it. Please do let me know if you want this to work according to a different convention.

File: testing/py3-pillow-simd/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+pkgrel=0
+_pypi_rel=0 # rare pypi "post-release" version used by this project
+pkgdesc="Python Imaging Library SIMD"
+url="https://github.com/uploadcare/pillow-simd"
+arch="x86_64"
+license="PSF"
+depends="py3-olefile freetype jpeg libwebp tiff libpng lcms2 libjpeg-turbo zlib"
+makedepends="python3-dev py3-setuptools freetype-dev jpeg-dev libwebp-dev
+	tiff-dev libpng-dev lcms2-dev libjpeg-turbo-dev zlib-dev"
+subpackages="py3-${_pkgname#py-}:_py3"
```
Comment (Cogitri) [inline testing/py3-pillow-simd/APKBUILD]: This seems obsolete now

File: testing/py3-pillow-simd/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+	mkdir -p "$pkgdir"
+}
+
+_py3() {
+	pkgdesc="$pkgdesc (for python3)"
+	install_if="$pkgname=$pkgver-r$pkgrel python3"
+
+	cd "$builddir"
+	python3 setup.py install --prefix=/usr --root="$subpkgdir"
+}
```
Comment (Cogitri) [inline testing/py3-pillow-simd/APKBUILD]: Same here, seems obsolete.

File: testing/py3-pillow-simd/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+depends="py3-olefile freetype jpeg libwebp tiff libpng lcms2 libjpeg-turbo zlib"
+makedepends="python3-dev py3-setuptools freetype-dev jpeg-dev libwebp-dev
+	tiff-dev libpng-dev lcms2-dev libjpeg-turbo-dev zlib-dev"
+subpackages="py3-${_pkgname#py-}:_py3"
+source="https://files.pythonhosted.org/packages/source/${_pkgname:0:1}/$_pkgname/$_pkgname-$pkgver.post$_pypi_rel.tar.gz"
+builddir="$srcdir/$_pkgname-$pkgver.post0"
+options="!check" # Unknown if test host has necessary SSE4 extensions.
+
+build() {
+	cd "$builddir"
```
Comment (Cogitri) [inline testing/py3-pillow-simd/APKBUILD]: No need to cd into $builddir anymore for stuff that isn't in main/

File: testing/py3-pillow-simd/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+pkgdesc="Python Imaging Library SIMD"
+url="https://github.com/uploadcare/pillow-simd"
+arch="x86_64"
+license="PSF"
+depends="py3-olefile freetype jpeg libwebp tiff libpng lcms2 libjpeg-turbo zlib"
+makedepends="python3-dev py3-setuptools freetype-dev jpeg-dev libwebp-dev
+	tiff-dev libpng-dev lcms2-dev libjpeg-turbo-dev zlib-dev"
+subpackages="py3-${_pkgname#py-}:_py3"
+source="https://files.pythonhosted.org/packages/source/${_pkgname:0:1}/$_pkgname/$_pkgname-$pkgver.post$_pypi_rel.tar.gz"
+builddir="$srcdir/$_pkgname-$pkgver.post0"
```
Comment (Cogitri) [inline testing/py3-pillow-simd/APKBUILD]: Shouldn't 0 be replaced with $_pypy_rel here?

File: testing/py3-pillow-simd/APKBUILD [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+pkgname=py3-pillow-simd
+_pkgname=Pillow-SIMD
+pkgver=6.0.0
+pkgrel=0
+_pypi_rel=0 # rare pypi "post-release" version used by this project
+pkgdesc="Python Imaging Library SIMD"
+url="https://github.com/uploadcare/pillow-simd"
+arch="x86_64"
+license="PSF"
+depends="py3-olefile freetype jpeg libwebp tiff libpng lcms2 libjpeg-turbo zlib"
```
Comment (Cogitri) [inline testing/py3-pillow-simd/APKBUILD]: Everything but py3-olefile is already detected automatically, see the build output starting at like 745: https://cloud.drone.io/alpinelinux/aports/11769/2/1



```

**Pull request closed without merging**